plugins/dfu/dfu-target.c

```diff
@@ -472,6 +474,52 @@ dfu_target_status_to_error_msg (DfuStatus status)
 	return NULL;
 }
 
+static gboolean
+dfu_target_poll_in_manifest (DfuTarget *target, GError **error)
+{
+	DfuTargetPrivate *priv = GET_PRIVATE (target);
+	DfuStatus status;
+	guint polling_count = 0;
+
+	/* get the status */
+	if (!dfu_device_refresh (priv->device, error))
+		return FALSE;
+
+	/* wait for DFU_STATE_DFU_MANIFEST to not be set */
+	while (dfu_device_get_state (priv->device)
+					== DFU_STATE_DFU_MANIFEST_SYNC ||
+			dfu_device_get_state (priv->device)
+					== DFU_STATE_DFU_MANIFEST) {
+		g_debug ("waiting for DFU_STATE_DFU_MANIFEST to clear");
+
+		if (polling_count++ > DFU_TARGET_MANIFEST_MAX_POLLING_TRIES) {
+			g_set_error_literal (error,
+			     FWUPD_ERROR,
+			     FWUPD_ERROR_INTERNAL,
+			     "reach to max polling tries");
+			return FALSE;
+		}
+
+		g_usleep ((dfu_device_get_download_timeout (
+					priv->device) + 1000) * 1000);
+		if (!dfu_device_refresh (priv->device, error))
+			return FALSE;
+	}
+
+	/* not in an error state */
+	if (dfu_device_get_state (priv->device) != DFU_STATE_DFU_ERROR)
+		return TRUE;
+
+	status = dfu_device_get_status (priv->device);
+	/* use a proper error description */
+	g_set_error_literal (error,
+			     FWUPD_ERROR,
+			     FWUPD_ERROR_INTERNAL,
+			     dfu_target_status_to_error_msg (status));
+
+	return FALSE;
```

Comment:
it might be nicer to flip this around, so that `dfu_device_get_state (priv->device) == DFU_STATE_DFU_ERROR` is the `if` block which returns FALSE and then we return TRUE at the end. Then status can move up into that scope too.

Reply:
I was just following the same coding style in dfu_target_check_status(). But I will change it as you suggested.